stdlib/2.7/__builtin__.pyi

```diff
@@ -529,7 +529,7 @@ class dict(MutableMapping[_KT, _VT], Generic[_KT, _VT]):
     @overload
     def __init__(self, map: Mapping[_KT, _VT], **kwargs: _VT) -> None: ...
     @overload
-    def __init__(self, iterable: Iterable[Tuple[_KT, _VT]], **kwargs: _VT) -> None: ...
+    def __init__(self, iterable: Iterable[Union[Tuple[_KT, _VT], Sequence[_VT]]], **kwargs: _VT) -> None: ...
```

Comment:
If we don't end up adding this case in typeshed (or even if we do), I think it would be useful to elaborate on that comment a bit, to make the situation clearer.  Something like
```
# NOTE: Keyword arguments are special.  If they are used, _KT must be a type
#       for which a str can be used, but we have no way to express that constraint
#       in the type annotations.  A type-checker may (and at least mypy does) have
#       a special case to enforce that constraint.
```

(That version also rewords the earlier part -- I find `_KT must include str` a confusing way of putting it.)